future-apps/payroll/contracts/Payroll.sol

```diff
@@ -652,11 +651,30 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
         if (accruedSalary > 0) {
             // Employee is cashing out a mixed amount between previous and current owed salaries;
             // first use up their accrued salary
-            employee.accruedSalary = uint256(0);
             // No need to use SafeMath here as we already know _paymentAmount > accruedSalary
             currentSalaryPaid = _paymentAmount - accruedSalary;
         }
-        employee.lastPayroll = _getLastPayrollDate(_employeeId, currentSalaryPaid);
+        uint256 salary = employee.denominationTokenSalary;
+        uint256 timeDiff = currentSalaryPaid.div(salary);
+
+        // If they're being paid an amount that doesn't match perfectly with the adjusted time
+        // (up to a seconds' worth of salary), add the second and put the extra remaining salary
+        // into their accrued salary
+        uint256 extraSalary = currentSalaryPaid % salary;
+        if (extraSalary > 0) {
+            timeDiff = timeDiff.add(1);
+            employee.accruedSalary = salary - currentSalaryPaid;
```

Comment:
@facuspagnuolo Not sure I understand this change in https://github.com/aragon/aragon-apps/pull/903/commits/ae3ee3d298a403fc894f5835ffc91a2c737ce562#diff-9d7adaffbcce19867f027a983a30bcc7R666 (Good catch on the `timeDiff` though!).

We've already calculated the remaining salary with `extraSalary`, and `salary - currentSalaryPaid` looks like it would underflow (`salary` is salary per second, `currentSalaryPaid` is some multiple of that).

Reply:
True.. I was thinking the case where you request less than a second of your salary, but ofc the scenario you're pointing out is also true. I think the tests are passing because we're using 1 USD per second which will always fit perfectly